Automattic/jetpack
# Boost: Add active modules to speed score request

Add Boost active modules data to page speed score request so that the same can be recorded as event properties while logging speed score data in Tracks.

#### Changes proposed in this Pull Request:

* Active modules are added to speed score request.
* The way this is achieved is by getting the list of active modules under [dispatch_speed_score_request](https://github.com/Automattic/jetpack/pull/21471/files#diff-d2294430df1afbcd40d748bc030bd228ea1a915b203368b0a330b23f5ee98e3bR94) and passing the list to the [Speed score request object](https://github.com/Automattic/jetpack/pull/21471/files#diff-d2294430df1afbcd40d748bc030bd228ea1a915b203368b0a330b23f5ee98e3bR94) and finally passing the modules list in the [request body](https://github.com/Automattic/jetpack/pull/21471/files#diff-d2294430df1afbcd40d748bc030bd228ea1a915b203368b0a330b23f5ee98e3bR94)

#### Jetpack product discussion
NA

#### Does this pull request change what data or activity we track or use?


Not at the moment. However, this PR sends the active modules list to the page speed score request.
The request to API happens as an async job. Update to the async job changes the data we track.
Reference to the async job update: D68731

#### Testing instructions:






* Setup WordPress.com sandbox
* Prepare your sandbox to test the API: FG /sandboxes/testing-apis-in-your-sandbox/
* Apply this patch in your sandbox: D68731
* Navigate to Jetpack Boost settings page: /wp-admin/admin.php?page=jetpack-boost
* Click the Refresh button under the "Overall score" section.
* Once the updated score is displayed, wait for 10 to 12 mins.
* Verify that the events `jetpack_boost_pagespeed_result` and `jetpack_boost_pagespeed_opportunity` has the event properties corresponding to the active modules in the following format: `module_critical_css: true`
